src/cheri/cheri-pte-ext.adoc

```diff
@@ -79,87 +83,101 @@ as usual, and capability reads are controlled by the CRG bit.
 
 * When a capability load or AMO instruction is executed, the implementation
  clears the {ctag} bit of the capability read from the virtual page.
-* When CRG is clear, the "no capability state", a {cheri_excep_name_pte_st}
- is raised when a capability store or AMO instruction is executed and the {ctag} bit
- of the capability being written is set.
-* When CRG is set, the "pre-CW state", two schemes are permitted (also see <<section_hardware_pte_updates>>):
-** The same behavior as when CRG is clear, allowing software interpretation
-of this state.
-** When a capability store or AMO instruction is executed
-and the {ctag} bit of the capability being written is set, the
-implementation sets the CW bit and assigns the CRG bit equal to <<sstatusreg_pte,sstatus>>.UCRG.
-+
-The PTE update must be
-atomic with respect to other accesses to the PTE, and must atomically check
-that the PTE is valid and grants sufficient permissions. Updates to the CW bit
-and CRG bit
-must be exact (i.e., not speculative), and observed in program order by the
-local hart. Furthermore, the PTE update must appear in the global memory order
-no later than the explicit memory access, or any subsequent explicit memory
-access to that virtual page by the local hart.  The ordering on loads and
-stores provided by FENCE instructions and the acquire/release bits on atomic
-instructions also orders the PTE updates associated with those loads and
-stores as observed by remote harts.
-+
-The PTE update is not required to be atomic with respect to the explicit memory
-access that caused the update, and the sequence is interruptible. However, the
-hart must not perform explicit memory access before the PTE update is globally
-visible.
+* When a capability store or AMO instruction is executed and the {ctag} bit
+ of the capability being written is set, the implementation raises a
+ {cheri_excep_name_pte_st}.
+
+When the CW bit is set, the "capability state", then the behaviour of
+capability loads is controlled by CRG and the behaviour of capability stores
+is controlled by CD.
 
 When CW is set, the CRG bit indicates the current generation of the virtual memory page with
 regards to the ongoing capability revocation cycle. Two schemes are permitted:
 
 * A {cheri_excep_name_pte_ld} is raised when a capability load or AMO instruction is executed
-with <<c_perm>> granted and the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG and PTE.U is set.
+ and:
+** the authorizing capability grants <<c_perm>>,
+** the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG, and
+** the virtual page's PTE.U is set.
 * A {cheri_excep_name_pte_ld} is raised when a capability load or AMO instruction is executed
-with <<c_perm>> granted and the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG and PTE.U is set
-and the capability read from memory has its tag set.
+ wherein:
```

Comment:
```suggestion
 and:
```

for consistency with the first scheme above

Reply:
in fact I think it's better not to list the first 3 points twice, and just add the optional check to the end of the first list.